typings/index.d.ts

```diff
@@ -214,7 +221,7 @@ export interface DateTimePickerProps {
 }
 
 export type ReactNativeModalDateTimePickerProps = DateTimePickerProps &
-  Omit<IOSNativeProps, "value" | "mode"> &
+  Omit<IOSNativeProps, "value" | "mode"> |
```

Comment:
This seems to work 👍 
Just to make it clearer, can we split this definition in two statements?
```js
type NativePickerProps = Omit<IOSNativeProps, "value" | "mode"> | Omit<AndroidNativeProps, "value" | "mode">

export type ReactNativeModalDateTimePickerProps = DateTimePickerProps & NativePickerProps;
```